nemo_skills/training/data_preparation_utils/preprocessing.py

```diff
@@ -397,3 +392,74 @@ def process(self):
                 samples_count += 1
 
         LOG.info("Prepared dataset size: %d", samples_count)
+
+
+class WriteFinalRLManifest(BaseProcessor):
+    def __init__(
+        self,
+        prompt_config: str,
+        prompt_template: str,
+        input_key: str = "input",
+        metadata: dict | None = None,
+        exclude_optional_keys: bool = True,
+        random_seed: int = 0,
+        do_shuffle: bool = True,
+        num_output_samples: int | None = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.input_key = input_key
+        self.metadata = metadata
+        self.exclude_optional_keys = exclude_optional_keys
+        if not self.metadata:
+            self.metadata = {}
+
+        self.prompt = None
+        if prompt_config and prompt_template:
+            self.prompt = get_prompt(prompt_config, prompt_template)
+        else:
+            LOG.warning("Prompt details are missing! The processed data won't be formatted using any prompt.")
+
+        self.random_seed = random_seed
+        self.do_shuffle = do_shuffle
+        self.num_output_samples = num_output_samples
+
+    def process(self):
+        samples_count = 0
+        all_data = []
+        with (open(self.input_manifest_file, "rt", encoding="utf-8") as fin,):
+            # only looping over the correct samples (unless asked for incorrect)
+            for line in fin:
+                elem = json.loads(line)
+                if 'expected_answer' in elem:
+                    elem['expected_answer'] = str(elem['expected_answer'])
+                # take only required keys from the input if exclude_optional_keys is True
+                output_sample = {}
+                if not self.exclude_optional_keys:
+                    output_sample = json.loads(line)
```

Comment:
This `if/elif` structure is testing for two very different things. Is there a missing `else` for the first `if`?  Can the second `elif` be an `if` in itself?